armory/metrics/task.py

```diff
@@ -1311,6 +1311,210 @@ def carla_od_AP_per_class(y_list, y_pred_list, iou_threshold=0.5, mean=True):
     )
 
 
+def _object_detection_poisoning_get_targeted_mr_dr(
+    y_list, y_pred_list, source_class=None, target_class=None, iou_threshold=0.5, score_threshold=0.5,
```

Comment:
just so I understand, what's prompted the need for a new function rather than using the disappearance/misclassification rate code we already have?

Reply:
Mainly the targeted nature of this metric.  For regional misclassification, we only count a box if the gt label is source and it was predicted as target.  For global misclassification, it only counts if the gt label is not target but it is predicted as target.  The existing code did not support this.  

For disappearance, it would be possible to use the existing code with ```class_list=source```, but it does something slightly different than my interpretation of what this metric is trying to capture.  I simply want to count source class boxes with no overlapping predicted box (over the threshold).  The existing code uses ```1 - TPR - misclassification_rate```.  I believe this could result in too low a disappearance rate if a GT box has multiple overlapping predictions.  For example, imagine two GT boxes, one with no prediction, and one with two predictions of which one is correct and one is wrong.  Disappearance rate would be 0 even though one GT box had nothing predicted.

Although a comment in the original code states ```Convert these arrays to binary to avoid double-counting (i.e. when multiple predicted boxes overlap with a single ground-truth box)```, this only avoids double counting two misclassifications or two TPs, not one of each.

I am happy to discuss this more or change my view if I'm missing something.